src/Enterspeed.Source.UmbracoCms/Services/DataProperties/DefaultConverters/DefaultBlockGridPropertyValueConverter.cs

```diff
@@ -0,0 +1,122 @@
+﻿#if NET7_0_OR_GREATER
```

Comment:
I've made some changes to get the properties for the layout blocks them self incl. `ariaGridColumn`. Until know we have not included the properties for the layout block it self, but only the blocks inside the areas. 
Also I uncluded the `gridColumns` on the root element as you would need this info to create the grid before you start to render the items inside the grid.

We might still miss some info, as I seen the UYmbraco documentation mention something about `ForceLeft `and `ForceRight` and also each area has an alias the we might need as well.

```suggestion
#if NET7_0_OR_GREATER
using System;
using System.Collections.Generic;
using System.Linq;
using Enterspeed.Source.Sdk.Api.Models.Properties;
using Enterspeed.Source.UmbracoCms.Extensions;
using Microsoft.Extensions.DependencyInjection;
using Umbraco.Cms.Core.Models.Blocks;
using Umbraco.Cms.Core.Models.PublishedContent;

namespace Enterspeed.Source.UmbracoCms.Services.DataProperties.DefaultConverters
{
    public class DefaultBlockGridPropertyValueConverter : IEnterspeedPropertyValueConverter
    {
        private readonly IServiceProvider _serviceProvider;

        public DefaultBlockGridPropertyValueConverter(IServiceProvider serviceProvider)
        {
            _serviceProvider = serviceProvider;
        }

        public bool IsConverter(IPublishedPropertyType propertyType)
        {
            return propertyType.EditorAlias.Equals("Umbraco.BlockGrid");
        }

        public virtual IEnterspeedProperty Convert(IPublishedProperty property, string culture)
        {
            var value = property.GetValue<BlockGridModel>(culture);
            var blockGridProperties = new Dictionary<string, IEnterspeedProperty>();
            var arrayItems = new List<IEnterspeedProperty>();

            if (value != null)
            {
                // NOTE: This needs to be resolved manually, since it would cause a circular dependency if injected through constructor
                var dataPropertyService = _serviceProvider.GetRequiredService<IEnterspeedPropertyService>();
                
                if (value.GridColumns != null)
                {
                    blockGridProperties.Add("gridColumns", new NumberEnterspeedProperty("gridColumns", value.GridColumns.Value));
                }

                foreach (var item in value)
                {
                    if (item.Areas.Any())
                    {
                        var areas = AddAreas(item, dataPropertyService, culture);
                        arrayItems.Add(areas);
                    }
                    else
                    {
                        var properties = MapProperties(dataPropertyService, item, culture);
                        arrayItems.Add(new ObjectEnterspeedProperty(properties));
                    }
                }

                blockGridProperties.Add("items", new ArrayEnterspeedProperty("items", arrayItems.ToArray()));
            }

            return new ObjectEnterspeedProperty(property.Alias, blockGridProperties);
        }

        private static ObjectEnterspeedProperty AddAreas(BlockGridItem item, IEnterspeedPropertyService dataPropertyService, string culture)
        {
            var blockGridAreaProperties = new Dictionary<string, IEnterspeedProperty>();
            var areasArrayItems = new List<IEnterspeedProperty>();

            if (item.Areas.Any())
            {
                blockGridAreaProperties = MapProperties(dataPropertyService, item, culture);

                foreach (var area in item.Areas)
                {
                    foreach (var blockGridItem in area)
                    {
                        var properties = MapProperties(dataPropertyService, blockGridItem, culture);

                        if (blockGridItem.Areas.Any())
                        {
                            properties.Add("areas", AddAreas(blockGridItem, dataPropertyService, culture));
                        }

                        areasArrayItems.Add(new ObjectEnterspeedProperty(properties));
                    }
                }
                
                blockGridAreaProperties.Add("areas", new ArrayEnterspeedProperty("areas", areasArrayItems.ToArray()));
            }

            return new ObjectEnterspeedProperty(blockGridAreaProperties);
        }

        private static Dictionary<string, IEnterspeedProperty> MapProperties(IEnterspeedPropertyService dataPropertyService, BlockGridItem item, string culture)
        {
            var properties = new Dictionary<string, IEnterspeedProperty>();

            if (item.Content?.Properties != null)
            {
                var contentProperties = dataPropertyService.ConvertProperties(item.Content.Properties, culture);
                properties.Add("content", new ObjectEnterspeedProperty(contentProperties));
            }

            if (item.Settings?.Properties != null)
            {
                var settingsProperties = dataPropertyService.ConvertProperties(item.Settings.Properties, culture);
                properties.Add("settings", new ObjectEnterspeedProperty(settingsProperties));
            }

            if (item.Content?.ContentType != null)
            {
                properties.Add("contentType", new StringEnterspeedProperty(item.Content.ContentType.Alias));
            }

            if (item.AreaGridColumns.HasValue)
            {
                properties.Add("areaGridColumns", new NumberEnterspeedProperty(item.AreaGridColumns.Value));
            }

            if (item.GridColumns.HasValue)
            {
                properties.Add("gridColumns", new NumberEnterspeedProperty(item.GridColumns.Value));
            }

            properties.Add("columnSpan", new NumberEnterspeedProperty(item.ColumnSpan));
            properties.Add("rowSpan", new NumberEnterspeedProperty(item.RowSpan));

            return properties;
        }
    }
}
#endif
```

The result will look something like this:

```
"blockGrid": {
	"gridColumns": 12,
	"items": [
		{
			"content": {
				"headline": "My headline"
			},
			"settings": {
				"hide": true
			},
			"contentType": "umbBlockGridDemoHeadlineBlock",
			"gridColumns": 12,
			"columnSpan": 4,
			"rowSpan": 2
		},
		{
			"content": {
				"layoutTitle": ""
			},
			"contentType": "umbBlockGridDemoTwoColumnLayoutBlock",
			"areaGridColumns": 10,
			"gridColumns": 12,
			"columnSpan": 12,
			"rowSpan": 1,
			"areas": [
				{
					"content": {
						"image": [
							{
								"id": 1067,
								"url": "https://localhost:44328/media/f5cdfcyd/how-to-write-a-good-blog-post.png"
							}
						]
					},
					"settings": {
						"hide": true
					},
					"contentType": "umbBlockGridDemoImageBlock",
					"gridColumns": 12,
					"columnSpan": 5,
					"rowSpan": 1
				},
				{
					"content": {
						"richText": "<p>My rich text content</p>"
					},
					"contentType": "umbBlockGridDemoRichTextBlock",
					"gridColumns": 12,
					"columnSpan": 5,
					"rowSpan": 1
				}
			]
		},
		{
			"content": {
				"layoutTitle": "test"
			},
			"contentType": "umbBlockGridDemoTwoColumnLayoutBlock",
			"areaGridColumns": 10,
			"gridColumns": 12,
			"columnSpan": 12,
			"rowSpan": 1,
			"areas": [
				{
					"content": {
						"headline": "right title"
					},
					"settings": {
						"hide": false
					},
					"contentType": "umbBlockGridDemoHeadlineBlock",
					"gridColumns": 12,
					"columnSpan": 4,
					"rowSpan": 1
				}
			]
		}
	]
}
```

Reply:
I will look into the potential missing properties. Gotta be honest, i used the sample data from the starter site to look for properties. Looks like that this is not a valid example to ensure that all properties are included. 

Regarding alias on area items. This should already be added. :) 

![image](https://user-images.githubusercontent.com/61312252/210446794-c726fef3-e488-412a-9603-c2f4e74ed9a5.png)
![image](https://user-images.githubusercontent.com/61312252/210446922-24e1d69c-06d2-4e6c-bad0-922a51ebd9f0.png)

Reply:
` json {
	"sourceId": "gid://Source/93029b08-c6f5-4656-8d94-96d7ca163bc6",
	"id": "gid://Source/93029b08-c6f5-4656-8d94-96d7ca163bc6/Entity/1124-en-us",
	"type": "contentPage",
	"originId": "1124-en-us",
	"originParentId": "1102-en-us",
	"url": "/about-us/",
	"redirects": [],
	"properties": {
		"seoMetaDescription": "",
		"keywords": [],
		"umbracoNavihide": false,
		"pageTitle": "About Us",
		"bodyText": {
			"items": [
				{
					"content": {},
					"contentType": "umbBlockGridDemoTwoColumnLayoutBlock",
					"columnSpan": 12,
					"rowSpan": 1,
					"areas": [
						{
							"alias": "left",
							"items": [
								{
									"content": {
										"headline": "test"
									},
									"contentType": "umbBlockGridDemoHeadlineBlock",
									"columnSpan": 6,
									"rowSpan": 1
								},
								{
									"content": {
										"headline": "testtt"
									},
									"contentType": "umbBlockGridDemoHeadlineBlock",
									"columnSpan": 6,
									"rowSpan": 1
								}
							]
						},
						{
							"alias": "right",
							"items": [
								{
									"content": {
										"headline": "Secondary headline"
									},
									"contentType": "umbBlockGridDemoHeadlineBlock",
									"columnSpan": 6,
									"rowSpan": 1
								}
							]
						}
					]
				},
				{
					"content": {
						"headline": "Headline"
					},
					"contentType": "umbBlockGridDemoHeadlineBlock",
					"columnSpan": 12,
					"rowSpan": 1
				}
			]
		},
		"metaData": {
			"name": "About Us",
			"culture": "en-us",
			"domain": null,
			"sortOrder": 2,
			"level": 2,
			"createDate": "2022-12-23T18:02:19",
			"updateDate": "2023-01-03T22:58:34",
			"nodePath": [
				"1102-en-us",
				"1124-en-us"
			]
		}
	}
}`

Reply:
<img width="906" alt="image" src="https://user-images.githubusercontent.com/61312252/210487667-828663f3-1bcc-46f3-9675-af1eb8121a0e.png">

Output after latest changes.

Reply:
@jesperweber  Could you verify the latest changes that has been pushed?

Reply:
I have tested and it's looking good :-)